src/output-json-arp.c

```diff
@@ -0,0 +1,254 @@
+
+#include "suricata-common.h"
+#include "detect.h"
+#include "flow.h"
+#include "conf.h"
+
+#include "threads.h"
+#include "tm-threads.h"
+#include "threadvars.h"
+#include "util-debug.h"
+
+#include "decode-ipv4.h"
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-reference.h"
+
+#include "output.h"
+#include "output-json.h"
+#include "output-json-arp.h"
+
+#include "util-classification-config.h"
+#include "util-privs.h"
+#include "util-print.h"
+#include "util-proto-name.h"
+#include "util-logopenfile.h"
+#include "util-time.h"
+#include "util-buffer.h"
+
+static const char *HwtypeToString(uint16_t hwtype)
+{
+    switch (hwtype) {
+        case 0:
+            return "reserved";
+        case 1:
+            return "ethernet";
+        case 2:
+            return "experimental_ethernet";
+        case 3:
+            return "amateur_radio_ax25";
+        case 4:
+            return "proteon_token_ring";
+        case 5:
+            return "chaos";
+        case 6:
+            return "ieee802";
+        case 7:
+            return "arcnet";
+        case 8:
+            return "hyperchannel";
+        case 9:
+            return "lanstar";
+        case 10:
+            return "autonet_short_address";
+        case 11:
+            return "localtalk";
+        case 12:
+            return "localnet";
+        case 13:
+            return "ultralink";
+        case 14:
+            return "smds";
+        case 15:
+            return "framerelay";
+        case 16:
+        case 19:
+        case 21:
+            return "atm";
+        case 17:
+            return "hdlc";
+        case 18:
+            return "fibre_channel";
+        case 20:
+            return "serial_line";
+        case 22:
+            return "mil_std_188_220";
+        case 23:
+            return "metricom";
+        case 24:
+            return "ieee1394.1995";
+        case 25:
+            return "mapos";
+        case 26:
+            return "twinaxial";
+        case 27:
+            return "eui64";
+        case 28:
+            return "hiparp";
+        case 29:
+            return "ip_over_arp";
+        case 30:
+            return "arpsec";
+        case 31:
+            return "ipsec";
+        case 32:
+            return "infiniband";
+        case 33:
+            return "cai";
+        case 34:
+            return "wiegand_interface";
+        case 35:
+            return "pure_ip";
+        case 36:
+            return "hw_exp1";
+        case 37:
+            return "hw_exp2";
+        case 38:
+            return "unified_bus";
+        case 256:
+            return "hw_exp2";
+        case 257:
+            return "aethernet";
+        default:
+            return "unknown";
+    }
+}
+
+static const char *PrototypeToString(uint16_t proto)
+{
+    switch (proto) {
+        case 0x0800:
+            return "ipv4";
+        default:
+            return "unknown";
+    }
+}
+
+static const char *OpcodeToString(uint16_t opcode)
```

Comment:
I've reviewed the kernel code for ARP [1] and noted that it supports only a few operations, as listed below:
```
/* ARP protocol opcodes. */
#define	ARPOP_REQUEST	1		/* ARP request			*/
#define	ARPOP_REPLY	2		/* ARP reply			*/
#define	ARPOP_RREQUEST	3		/* RARP request			*/
#define	ARPOP_RREPLY	4		/* RARP reply			*/
#define	ARPOP_InREQUEST	8		/* InARP request		*/
#define	ARPOP_InREPLY	9		/* InARP reply			*/
#define	ARPOP_NAK	10		/* (ATM)ARP NAK			*/
```

I plan to do the same and remove the unsupported operations.

[1]  https://github.com/torvalds/linux/blob/master/include/uapi/linux/if_arp.h#L31

Reply:
Why not leave this? It's useful in output?

Reply:
I believe logging the operation type would be helpful.